source/material_model/visco_plastic.cc

```diff
@@ -0,0 +1,617 @@
+          prefactors_diffusion = get_vector_double("Prefactors for diffusion creep", n_fields, prm);
+          stress_exponents_diffusion = get_vector_double("Stress exponents for diffusion creep", n_fields, prm);
+          grain_size_exponents_diffusion = get_vector_double("Grain size exponents for diffusion creep", n_fields, prm);
+          activation_energies_diffusion = get_vector_double("Activation energies for diffusion creep", n_fields, prm);
+          activation_volumes_diffusion = get_vector_double("Activation volumes for diffusion creep", n_fields, prm);
+          // Dislocation creep parameters (Note the lack of grain size exponents)
+          prefactors_dislocation = get_vector_double("Prefactors for dislocation creep", n_fields, prm);
+          stress_exponents_dislocation = get_vector_double("Stress exponents for dislocation creep", n_fields, prm);
+          activation_energies_dislocation = get_vector_double("Activation energies for dislocation creep", n_fields, prm);
+          activation_volumes_dislocation = get_vector_double("Activation volumes for dislocation creep", n_fields, prm);
+          // Plasticity parameters
+          angles_internal_friction = get_vector_double("Angles of internal friction", n_fields, prm);
+          cohesions = get_vector_double("Cohesions", n_fields, prm);
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = NonlinearDependence::temperature | NonlinearDependence::pressure | NonlinearDependence::strain_rate | NonlinearDependence::compositional_fields;
+      this->model_dependence.density = NonlinearDependence::temperature | NonlinearDependence::pressure | NonlinearDependence::compositional_fields;
+      this->model_dependence.compressibility = NonlinearDependence::none;
+      this->model_dependence.specific_heat = NonlinearDependence::none;
+      this->model_dependence.thermal_conductivity = NonlinearDependence::temperature | NonlinearDependence::pressure | NonlinearDependence::compositional_fields;
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(ViscoPlastic,
+                                   "visco plastic",
+                                   "An implementation of a visco-plastic rheology with options for "
+                                   "selecting dislocation creep, diffusion creep or composite "
+                                   "viscous flow laws.  Plasticity limits viscous stresses through "
+                                   "a Drucker Prager yield criterion. The model is incompressible. "
+                                   "Note that this material model is based heavily on the "
+                                   "DiffusionDislocation (Rene Gassmoeller) and DruckerPrager "
```

Comment:
Also, I believe Bob Myhill (@bobmyhill) wrote the diffusion dislocation material model originally, and Anne Glerum (@anne-glerum) contributed the drucker prager model.